packages/seacas/libraries/ioss/src/text_mesh/Iotm_TextMesh.C

```diff
@@ -0,0 +1,626 @@
+//
+// See packages/seacas/LICENSE for details
+
+#include "Iotm_TextMesh.h"
+
+#include <Ioss_Utils.h>
+#include <fmt/ostream.h>
+#include <sys/types.h>  // for ssize_t
+#include <tokenize.h>   // for tokenize
+
+#include <algorithm>
+#include <cassert>  // for assert
+#include <cmath>    // for atan2, cos, sin
+#include <cstdlib>  // for nullptr, exit, etc
+#include <iostream>
+#include <numeric>
+#include <string>
+#include <vector>  // for vector
+#include <array>
+
+#include <Ioss_Utils.h>
+
+#define ThrowRequireMsg(expr, message)                                       \
+  do {                                                                       \
+    if ( !(expr) ) {                                                         \
+      std::ostringstream internal_throw_require_oss;                         \
+      internal_throw_require_oss << message;                                 \
+      IOSS_ERROR(internal_throw_require_oss );                               \
+    }                                                                        \
+  } while (false)
+
+namespace Iotm
+{
+void error_handler(const std::ostringstream& message)
+{
+  IOSS_ERROR(message);
+}
+
+TextMesh::TextMesh(int proc_count, int my_proc) : m_processorCount(proc_count), m_myProcessor(my_proc)
+{
+  initialize();
+}
+
+TextMesh::TextMesh(const std::string &parameters, int proc_count, int my_proc)
+    : m_processorCount(proc_count), m_myProcessor(my_proc)
+{
+  // Possible that the 'parameters' has the working directory path
+  // prepended to the parameter list.  Strip off everything in front
+  // of the last '/' (if any)...
+  auto params = Ioss::tokenize(parameters, "/");
+
+  if (!params.empty()) {
+    auto groups = Ioss::tokenize(params.back(), "|+");
+    parse_options(groups);
+
+    auto errorHandler = [] (const std::ostringstream& errmsg) {
+                               error_handler(errmsg);
+                             };
+
+    TextMeshParser parser(m_dimension);
+    parser.set_error_handler(errorHandler);
+    m_data = parser.parse(groups[0]);
+
+    m_coordinates.set_error_handler(errorHandler);
+    m_coordinates.set_coordinate_data(m_data, m_rawCoordinates);
+  };
+
+  initialize();
+}
+
+TextMesh::TextMesh()
+{
+  initialize();
+}
+
+unsigned TextMesh::spatial_dimension() const
+{
+  return m_dimension;
+}
+
+void TextMesh::initialize()
+{
+
+  build_part_to_topology_map();
+  build_block_partition_map();
+  build_element_connectivity_map();
+
+  m_variableCount[Ioss::COMMSET] = 0;
+  m_variableCount[Ioss::ELEMENTBLOCK] = 0;
+  m_variableCount[Ioss::INVALID_TYPE] = 0;
+  m_variableCount[Ioss::NODEBLOCK] = 0;
+  m_variableCount[Ioss::REGION] = 0;
+
+}
+
+void TextMesh::parse_coordinates_option(const std::vector<std::string> &option)
+{
+  if (option.size() > 1) {
+    auto tokens = Ioss::tokenize(option[1], ",");
+    m_rawCoordinates.reserve(tokens.size());
+    for (auto token : tokens) {
```

Comment:
Sure, easy fix. For the most part, Timothy had insisted that TextMesh would only be used for small meshes so that performance was not of paramount concern .... but I'm sure there is that one user out there that will abuse this ;)